src/appleseed/foundation/math/sampling/mappings.h

```diff
@@ -558,7 +558,7 @@ inline T exponential_distribution_on_segment_pdf(
     const T l,
     const T r)
 {
-    assert (x >= l && x < r);
+    assert (x >= l && x <= r);
```

Comment:
Are you sure the upper bound should be included? If yes, you'll need to update the function's description.

Reply:
I am clamping at line 551.